tests/test_patterns.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2010 Yesudeep Mangalapilly <yesudeep@gmail.com>
+# Copyright 2020 Boris Staletic <boris.staletic@gmail.com>
+
+import pytest
+from pathlib import PureWindowsPath, PurePosixPath
+from watchdog.utils.patterns import _match_path, filter_paths, match_any_paths
+
+
+@pytest.mark.parametrize("input,included_patterns,excluded_patterns,expected", [
+    (PurePosixPath("/users/gorakhargosh/foobar.py"), ["*.py"], ["*.PY"], True),
+    (PurePosixPath("/users/gorakhargosh/foobar.py"), ["*.py"], ["*.PY"], True),
+    (PureWindowsPath("/users/gorakhargosh/foobar.py"), ["*.py"], ["*.PY"], False),
+    (PureWindowsPath("/users/gorakhargosh/foobar.py"), ["*.py"], ["*.PY"], ValueError),
```

Comment:
You should use `Path` here, right? Users will not use specific class but directly `Path`.

Reply:
No. `_match_path` is implementation detail and, as the leading underscore suggests, is "private". It's used by the other two functions - `filter_paths` and `match_any_paths`, which convert the user's input (be it a string or a `Path`) to the respective pure paths. Thus `_match_path` doesn't care about case-sensitivity and expects that to be "chosen" by the right `PurePosixPath`/`PureWindowsPath` input.

Reply:
I stand corrected 👍